homeassistant/components/template/coordinator.py

```diff
@@ -91,16 +101,68 @@ async def _attach_triggers(self, start_event=None) -> None:
             start_event is not None,
         )
 
+    async def _setup_condition_checker(
+        self,
+    ) -> Callable[[TemplateVarsType | None], bool] | None:
+        cond_configs = self.config[CONF_CONDITION]
+        checks: list[condition.ConditionCheckerType] = [
+            await condition.async_from_config(self.hass, cond_config)
+            for cond_config in cond_configs
+        ]
+
+        def check_conditions(variables: TemplateVarsType | None = None) -> bool:
+            """AND all conditions."""
+            errors: list[ConditionErrorIndex] = []
+            for index, check in enumerate(checks):
+                try:
+                    with trace_path(["condition", str(index)]):
+                        if check(self.hass, variables) is False:
+                            return False
+                except ConditionError as ex:
+                    errors.append(
+                        ConditionErrorIndex(
+                            "condition", index=index, total=len(checks), error=ex
+                        )
+                    )
+
+            if errors:
+                _LOGGER.warning(
+                    "Error evaluating template trigger condition:\n%s",
+                    ConditionErrorContainer("condition", errors=errors),
+                )
+                return False
+
+            return True
+
+        return check_conditions
+
     async def _handle_triggered_with_script(self, run_variables, context=None):
+        if not await self._check_condition(run_variables, context):
+            return
         # Create a context referring to the trigger context.
         trigger_context_id = None if context is None else context.id
         script_context = Context(parent_id=trigger_context_id)
         if script_result := await self._script.async_run(run_variables, script_context):
             run_variables = script_result.variables
-        self._handle_triggered(run_variables, context)
+        self._execute_update(run_variables, context)
+
+    async def _handle_triggered(self, run_variables, context=None):
+        if not await self._check_condition(run_variables, context):
+            return
+        self._execute_update(run_variables, context)
+
+    async def _check_condition(self, run_variables, context=None) -> bool:
```

Comment:
It doesn't seem this needs to be a coroutine function?
```suggestion
    def _check_condition(self, run_variables, context=None) -> bool:
```